spec.html

```diff
@@ -32497,173 +32497,107 @@ <h1>String.prototype.replace ( _searchValue_, _replaceValue_ )</h1>
+        <emu-clause id="sec-getsubstitution" type="abstract operation" oldids="table-replacement-text-symbol-substitutions,table-45">
           <h1>
             GetSubstitution (
               _matched_: unknown,
               _str_: unknown,
               _position_: a non-negative integer,
               _captures_: unknown,
               _namedCaptures_: unknown,
-              _replacement_: unknown,
+              _replacer_: unknown,
             )
           </h1>
           <dl class="header">
           </dl>
+          <p>For the purposes of this abstract operation, a <em>decimal digit</em> is a code unit in the range 0x0030 (DIGIT ZERO) to 0x0039 (DIGIT NINE) inclusive.</p>
           <emu-alg>
             1. Assert: Type(_matched_) is String.
             1. Let _matchLength_ be the number of code units in _matched_.
             1. Assert: Type(_str_) is String.
             1. Let _stringLength_ be the number of code units in _str_.
             1. Assert: _position_ &le; _stringLength_.
-            1. Assert: _captures_ is a possibly empty List of Strings.
-            1. Assert: Type(_replacement_) is String.
+            1. Assert: _captures_ is a possibly empty List, each of whose elements is a String or *undefined*.
+            1. Assert: Type(_replacer_) is String.
+            1. Let _preceding_ be the substring of _str_ from 0 to _position_.
             1. Let _tailPos_ be _position_ + _matchLength_.
-            1. Let _m_ be the number of elements in _captures_.
-            1. Let _result_ be the String value derived from _replacement_ by copying code unit elements from _replacement_ to _result_ while performing replacements as specified in <emu-xref href="#table-replacement-text-symbol-substitutions"></emu-xref>. These `$` replacements are done left-to-right, and, once such a replacement is performed, the new replacement text is not subject to further replacements.
-            1. Return _result_.
+            1. If _tailPos_ &ge; _stringLength_, then
+              1. Let _following_ be the empty String.
+            1. Else,
+              1. Let _following_ be the substring of _str_ from _tailPos_.
+            1. Let _interpretNumericRef_ be a new Abstract Closure with parameters (_ref_) that captures _captures_ and performs the following steps when called:
+              1. Assert: _ref_ starts with *"$"*.
+              1. Let _digits_ be the substring of _ref_ from 1.
+              1. Assert: _digits_ consists of 1 or 2 decimal digits.
+              1. Let _index_ be the integer that results from interpreting _digits_ as a decimal numeral.
+              1. Assert: 0 &le; _index_ &le; 99.
+              1. If _index_ = 0 or _index_ &gt; the number of elements in _captures_, then
+                1. NOTE: No replacement is done.
+                1. Return _ref_.
+              1. Else,
+                1. Let _capture_ be the _index_<sup>th</sup> element of _captures_.
```

Comment:
Technically, I don't think it's ambiguous, since the definition of Lists sets up the equivalence between `the _n_<sup>th</sup> element of _L_` and `_L_[_n_ - 1]`. However, that's a fairly obscure point, so the suggested change is clearer.

It's a pre-existing condition, so I'll deal with it in a separate commit. I'm thinking I might also add a Note in RegExp.prototype [ @@replace ], maybe after 14.i.iii.

Reply:
I think it's plenty appropriate to change it in this PR.

Reply:
Yup, I meant a separate commit *in this PR*.